torchao/float8/fsdp_utils.py

```diff
@@ -59,17 +59,20 @@ def precompute_float8_dynamic_scale_for_fsdp(module: nn.Module) -> None:
         return
 
     # inf-norm is equivalent to max(abs(w))
-    max_weights = torch._foreach_norm(weights, ord=math.inf)  # Partial
+    # keep consistent with float8_utils.amax_to_scale
+    # torch.compile and eager show different numerics for 1.0 / float32,
+    # upcast to float64 to ensure same numeric between compile and eager
+    max_weights = torch._foreach_norm(weights, ord=math.inf, dtype=torch.float64)  # Partial
```

Comment:
`dtype=torch.float64` only changes the accumulation dtype? if there is no noticeable cost to this, I wonder if we should be doing this in more places 🤔

Reply:
good question! Actually I just updated the code to do `_foreach_norm` in original precision and do float64 upcasting before calculating scales. That ensures consistent implementation between prcompute and `float8_utils.amax_to_scale`

back to your question, I checked `ForeachReduceOp.cu` and it's dispatching to lpnorm_cleanup<scalar_t, NormType::LInf, out_t>. Not sure what's inside `lpnorm_cleanup`. But inf-norm is just max(abs) so not sure if they accumulate numerics
https://github.com/pytorch/pytorch/blob/a28b40fa74470058ca57d77652b9601bece2f4d5/aten/src/ATen/native/cuda/ForeachReduceOp.cu#L534-L535C19